### ISSUE TITLE
Re-enable rr for Pkg tests

### ISSUE THREAD

**Issue by Keno** [JuliaLang/julia#43521]
@DilumAluthge notes that rr is currently disabled when running the Pkg tests on CI because libgit2 blows up rr trace file size due to excessive syscall traffic. This is an upstream libgit2 issue: https://github.com/libgit2/libgit2/issues/6038. Opening this hear to record the knowledge and track the action item of updating the CI config once upstream libgit2 is fixed.

Our configuration: https://github.com/JuliaLang/julia/blob/f7280386af52cb708bccb8c6876a1df747af3651/.buildkite/pipelines/main/platforms/tester_linux.arches#L13

**Comment by vtjnash** [comment]
Can we build libgit2 with mmap disabled?

**Comment by Keno** [comment]
Potentially. Reading the upstream discussion, it appears that this used to be done with read/write but was changed to work around coherence issues mmaping network drives on Windows. So hopefully changing this back should be easy, though I don't think there's a build flag for it or anything.

**Comment by Keno** [comment]
Actually, https://github.com/libgit2/libgit2/pull/6039, may have already done exactly that. So we should probably try building with that and seeing if the issue is resolved.

**Comment by giordano** [comment]
That seems to be in 1.3.0: https://github.com/JuliaPackaging/Yggdrasil/pull/3962

**Comment by vtjnash** [comment]
Yeah, my suggestion was they should completely eliminate use of mmap everywhere, since read/write are usually much more optimal and reliable, particularly once adding threads. Let's see how it goes with that PR to start.

**Comment by vtjnash** [comment]
We could still build with `NO_MMAP`, but it seems that this is already tracked upstream in libgit2 (#6038), so we don't need to track it here

**Comment by Keno** [comment]
We've had to re-disable Pkg tests on CI. A brief investigation shows that the primary source of syscall traffic is a 20 byte read done using a full `mmap/munmap` inside `git_packfile_unpack_header`. I think this could be fixed in libgit2, but for now just writing down the observation.

**Comment by KristofferC** [comment]
This doesn't feel like it has to be on the milestone. Can be reenabled whenever libgit2 has better support for rr but I don't see why we should tie that to our release.
